cro-clib/src/transaction.rs

```diff
@@ -0,0 +1,318 @@
+use crate::types::get_string;
+use crate::types::CroStakedState;
+use crate::types::{CroAddress, CroAddressPtr, CroResult};
+use chain_core::init::coin::Coin;
+pub use chain_core::init::network::Network;
+use chain_core::state::account::{
+    StakedState, StakedStateAddress, StakedStateOpAttributes, StakedStateOpWitness, UnbondTx,
+    WithdrawUnbondedTx,
+};
+use chain_core::tx::data::access::{TxAccess, TxAccessPolicy};
+use chain_core::tx::data::address::ExtendedAddr;
+use chain_core::tx::data::attribute::TxAttributes;
+use chain_core::tx::data::output::TxOut;
+use chain_core::tx::TransactionId;
+use chain_core::tx::TxAux;
+use client_common::tendermint::types::AbciQueryExt;
+use client_common::tendermint::{Client, WebsocketRpcClient};
+use client_common::{ErrorKind, Result, ResultExt};
+use client_common::{PublicKey, SignedTransaction};
+use client_core::cipher::DefaultTransactionObfuscation;
+use client_core::TransactionObfuscation;
+use parity_scale_codec::Decode;
+use parity_scale_codec::Encode;
+use std::collections::BTreeSet;
+use std::os::raw::c_char;
+use std::ptr;
+use std::str::FromStr;
+use std::string::ToString;
+fn do_cro_unbond(
+    network: u8,
+    nonce: u64,
+    from_address: &CroAddress,
+    to_address_user: &str,
+    amount: u64,
+) -> Result<Vec<u8>> {
```

Comment:
just by looking at the name `do_cro_unbond`, the arguments and the return type `Result<Vec<u8>>` without any comments one cannot understand what this function is supposed to do:
1. `to_address_user: &str` -- what format is this supposed to be in? Hex?
2. `amount: u64` -- is this in base units? 
3. `Result<Vec<u8>>` -- when can this fail / return `Err`? What is in that `Vec<u8>`?
...

Even though these are internal functions (not exposed in the bindings), it'll be good to make them more maintainable -- e.g. add a comment explaining such things, give a better name (`create_encoded_signed_unbond`?), have more precise types for the arguments...

Reply:
ok